### PR TITLE
Remove WellStateFullyImplicitBlackoilDense

### PR DESCRIPTION
**Pull request by totto82**
After the restructuring of of the well model, keeping an extra class for
the "Dense" model is not needed. The only thing still left in
WellStateFullyImplicitBlackoilDense was some solvent related stuff, this
PR moves this to WellStateFullyImplicitBlackoil and removes
WellStateFullyImplicitBlackoilDense.

In addition to a cleaner code this PR fixes missing solvent well output.

### CODE REVIEW

**Comment by GitPaean** [comment]
jenkins build this please

File: opm/autodiff/BlackoilSolventModel_impl.hpp [hunk 1]
```diff
@@ -1098,7 +1098,8 @@ namespace Opm {
         if (has_solvent_ && is_miscible_ && b_eff_[0].size() == 0) {
             // A hack to avoid trouble for initial fluid in place, due to usage of b_eff_.
             WellState xw, xwdummy;
-            xw.init(&wells(), x, xwdummy);
+            const Opm::PhaseUsage pu = fluid_.phaseUsage();
```
Comment (GitPaean) [inline opm/autodiff/BlackoilSolventModel_impl.hpp]: reference please.

File: opm/autodiff/ParallelDebugOutput.hpp [hunk 2]
```diff
@@ -621,7 +621,7 @@ namespace Opm
                                            std::unordered_set<std::string>());
 
                 const Wells* wells = wells_manager.c_wells();
-                globalWellState_.init(wells, *globalReservoirState_, globalWellState_ );
+                globalWellState_.init(wells, *globalReservoirState_, globalWellState_,phaseUsage_ );
```
Comment (GitPaean) [inline opm/autodiff/ParallelDebugOutput.hpp]: space between arguments.

File: opm/autodiff/SimulatorBase_impl.hpp [hunk 3]
```diff
@@ -184,7 +184,7 @@ namespace Opm
                                        defunct_well_names_);
             const Wells* wells = wells_manager.c_wells();
             WellState well_state;
-            well_state.init(wells, state, prev_well_state);
+            well_state.init(wells, state, prev_well_state,props_.phaseUsage());
```
Comment (GitPaean) [inline opm/autodiff/SimulatorBase_impl.hpp]: space between arguments.

File: opm/autodiff/WellStateFullyImplicitBlackoil.hpp [hunk 4]
```diff
@@ -258,6 +277,7 @@ namespace Opm
                 assert(local_comp_index == this->wells_->well_connpos[ w + 1 ] - this->wells_->well_connpos[ w ]);
             }
 
+
```
Comment (GitPaean) [inline opm/autodiff/WellStateFullyImplicitBlackoil.hpp]: added line unnecessarily.

**Comment by totto82** [comment]
jenkins build this please

File: opm/autodiff/WellStateFullyImplicitBlackoilDense.hpp [hunk 5]
```diff
@@ -1,162 +0,0 @@
-/*
```
Comment (atgeirr) [inline opm/autodiff/WellStateFullyImplicitBlackoilDense.hpp]: This file must be removed from CMakeLists_files.cmake.

File: opm/autodiff/WellStateFullyImplicitBlackoil.hpp [hunk 6]
```diff
@@ -170,6 +174,18 @@ namespace Opm
                                 perfPress()[ perf ] = prevState.perfPress()[ oldPerf_idx ];
                             }
                         }
+                        // perfSolventRates
+
+                        if (pu.has_solvent) {
+                            if( num_perf_old_well == num_perf_this_well )
+                            {
+                                for (int perf = wells->well_connpos[ newIndex ];
+                                     perf < wells->well_connpos[ newIndex + 1]; ++perf, ++oldPerf_idx )
```
Comment (atgeirr) [inline opm/autodiff/WellStateFullyImplicitBlackoil.hpp]: I think there is a bug here: oldPerf_idx has already been used as a loop variable by the perfPressures loop above. The existing code was dangerous in that respect. Suggested fix: make oldPerf_idx a const variable on line 148 (new numbering) and use separate loop-scoped variables in the perfPressures and perfSolventRates loops.

Comment (totto82) [reply opm/autodiff/WellStateFullyImplicitBlackoil.hpp]: Good catch. Thanks.

**Comment by totto82** [comment]
jenkins build this please

**Pull request merged**